bootcompiler/src/main/scala/org/mozartoz/bootcompiler/transform/Desugar.scala

```diff
@@ -187,6 +187,25 @@ object Desugar extends Transformer with TreeDSL {
       treeCopy.MatchStatement(expr, value,
           clauses map(c => matchExpressionClauseToBindStatement(result, c)),
           exprToBindStatement(result, elseExpression))
+
+    case Record(label, fields) =>
+      val lastI = fields.lastIndexWhere(_.value match {
+        case CallExpression(callable, args) => true
+        case _                              => false
+      })
+      if (lastI < 0)
+        return treeCopy.BindStatement(expr, result, expr)
+      statementWithTemp { newVar =>
+        treeCopy.CompoundStatement(expr, Seq(
+          result === Record(label, fields.indices.map(i => {
+            val field = fields(i)
+            treeCopy.RecordField(field, field.feature, if (i == lastI) newVar else field.value)
+          })),
+          fields(lastI).value match {
+            case CallExpression(callable, args) =>
+              treeCopy.CallStatement(fields(lastI), callable, Simplify.putVarInArgs(args, newVar))
+          }))
+      }
```

Comment:
I think a lot of this could be simplified if we just use elements instead of indices.
You could then match it with `map(f => { if (f eq lastCall)`